### PR TITLE
SOG Compat - Add `isDamageAllowed` check to punji damage handler

### PR DESCRIPTION
**Pull request by Zman6258**
**When merged this pull request will:**
- Title.

### CODE REVIEW

File: optionals/compat_sog/functions/fnc_handlePunjiTrapDamage.sqf [hunk 1]
```diff
@@ -19,7 +19,7 @@ params ["_trap"];
 if (isNull (configFile >> "CfgPatches" >> "ace_medical")) exitWith {};
 
 private _radius = getNumber (configOf _trap >> "indirectHitRange");
-private _affectedUnits = (_trap nearEntities ["CAManBase", _radius]) select {local _x};
+private _affectedUnits = (_trap nearEntities ["CAManBase", _radius]) select {local _x && isDamageAllowed _x};
```
Comment (commy2) [inline optionals/compat_sog/functions/fnc_handlePunjiTrapDamage.sqf:22]: Maybe "lazy select", i.e. 
```suggestion
private _affectedUnits = (_trap nearEntities ["CAManBase", _radius]) select {local _x} select {isDamageAllowed _x};
```

Comment (Zman6258) [reply optionals/compat_sog/functions/fnc_handlePunjiTrapDamage.sqf:22]: Oh cool, I never thought of doing it like that

**Comment by veteran29** [comment]
Could use the new parameter introduced in #8599 instead of the filter.

**Comment by PabstMirror** [comment]
right now 8599 throws an `ERROR_1` (we might want to change?), 
but I still think it's better to handle in the calling function like this PR is doing

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**